### PR TITLE
REF Make Moran’s I and Geary’s C efficient standard morphology features

### PR DESCRIPTION
**Pull request by hubertgabrys**
Moran’s I and Geary’s C previously constructed dense pairwise matrices, making large CT ROIs impractical to process. Both measures now share an exact calculation using hybrid FFT/Parseval sums or blocked SciPy pair sums, selected automatically from voxel count and bounding-box geometry.

Both features are included in standard morphology extraction and default API/GUI output. The dedicated `morphology_correlation` family is removed; individual feature-name selection remains available.

- Preserves full inverse physical-distance weights without neighborhood truncation or sampling.
- Introduces no resource caps or additional dependencies.
- Updates the user guide and API documentation.

Validation covers direct numerical comparisons, anisotropic spacing, irregular masks, re-segmentation, edge cases, feature selection, and GUI CSV output. Both calculation paths pass IBSI-I reference tolerances for configurations A–E. All 173 relevant tests passed, along with lint and whitespace checks.

### CODE REVIEW

File: zrad/radiomics/morphology.py [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+    """Choose exact spatial evaluation from valid count and padded geometry.
+
+    Keep geometry separate from evaluation: the same N and FFT shape can be
+    used by a future general extraction policy without changing either algorithm.
+    """
+    padded_size = math.prod(fft_shape)
+    pairs = n * (n - 1) // 2
+    fft_cost = _CORRELATION_FFT_COST_FACTOR * padded_size * math.log2(max(padded_size, 2))
+    if n > _CORRELATION_TINY_ROI and pairs > fft_cost:
+        return 'fft'
```
[the remark this answers is not in the document]
Comment (hubertgabrys) [reply zrad/radiomics/morphology.py:29]: Resource policy for Z-Rad will be introduced (if ever) in a separate PR

**Review by maksymfritsak — APPROVED** [review]
Amazing, thanks, Hubert!

When I talked to Hero Imaging (paid radiomics software) at ESTRO this year, I asked whether they had implemented these two features. They told me they had omitted them completely—another reason why Z-Rad is the best!
